app/models/metric/ci_mixin/capture.rb

```diff
@@ -72,8 +71,12 @@ def perf_capture_queue(interval_name, options = {})
     }
 
     items.each do |item|
+      # Should both interval name and args (dates) be part of uniqueness query?
+      item_interval = item.shift
```

Comment:
This can be simplified a bit.

```ruby
     items.each do |item_interval, *item|
       # item_interval = item.shift <<< Don't need this anymore
```

This way we avoid mutating the block argument via `shift`ing.

Also, the variables could be named slightly differently to be more consistent with the code defined above. What about renaming `item_interval` to `interval_name` and `item` to `args` or `start_and_end_time`?

```ruby
    items.each do |interval_name, *start_and_end_time|
      # Should both interval name and args (dates) be part of uniqueness query?
      queue_item_options = queue_item.merge(:method_name => "perf_capture_#{interval_name}")
      queue_item_options[:args] = start_and_end_time if start_and_end_time.present?
# ...
```

`items` is also way too vague (`intervals` could be more descriptive I suppose), but renaming it is probably out of scope of this PR.